modules/schema/src/main/java/org/apache/ignite/internal/schema/configuration/TableValidatorImpl.java

```diff
@@ -41,10 +42,9 @@ public void validate(TableValidator annotation, ValidationContext<NamedListView<
             TableView newTable = newTables.get(tableName);
 
             try {
-                TableDefinitionImpl tbl = SchemaConfigurationConverter.convert(newTable);
-
-                assert !tbl.keyColumns().isEmpty();
-                assert !tbl.colocationColumns().isEmpty();
+                assert newTable.primaryKey() != null;
```

Comment:
why asserts acceptable here ?

Reply:
This is straightforward replace of definition to view

Reply:
Rework this part anyway